storm/check.py

```diff
@@ -0,0 +1,573 @@
+            'env:{}'.format(self.environment_name),
+            'environment:{}'.format(self.environment_name)] + additional_tags
+        self.histogram(
+            metric=metric,
+            value=value,
+            tags=all_tags
+        )
+
+    def update_from_config(self, instance):
+        """ Update Configuration tunables from instance configuration.
+
+        :param instance: Agent config instance.
+        :return: None
+        """
+        self.nimbus_server = instance.get('server', self.init_config.get('server', 'http://localhost:9005'))
+        self.environment_name = instance.get('environment', self.init_config.get('environment', 'dev'))
+        self.additional_tags.extend(instance.get('tags', []))
+        self.excluded_topologies.extend(instance.get('excluded', []))
+        intervals = instance.get('intervals', self.init_config.get('intervals', [60]))
+
+        if not isinstance(intervals, (list, tuple)) or len(intervals) < 1:
+            raise AssertionError("Expected intervals to be a list of integers with at least 1 value")
+        else:
+            self.intervals.extend(intervals)
+
+    def check(self, instance):
+        """ Perform the agent check.
+
+        :param instance: Agent instance.
+        :return: None
+        """
+        # Setup
+        self.update_from_config(instance)
+
+        # Cluster Stats
+        cluster_stats = self.get_storm_cluster_summary()
+        self.process_cluster_stats(self.environment_name, cluster_stats)
+
+        # Nimbus Stats
+        nimbus_stats = self.get_storm_nimbus_summary()
+        self.process_nimbus_stats(self.environment_name, nimbus_stats)
+
+        # Supervisor Stats
+        supervisor_stats = self.get_storm_supervisor_summary()
+        self.process_supervisor_stats(supervisor_stats)
+
+        # Topology Stats
+        summary = self.get_storm_topology_summary()
+        for topology in summary['topologies']:
+            topology_id = topology.get('id')
+            topology_name = _g(topology, 'unknown', None, 'name')
+            topology_status = None
+            if topology_name not in self.excluded_topologies:
+                for interval in self.intervals:
+                    stats = self.get_topology_info(topology_id=topology_id, interval=interval)
+                    self.process_topology_stats(topology_stats=stats, interval=interval)
+                    metric_stats = self.get_topology_metrics(topology_id=topology_id, interval=interval)
+                    self.process_topology_metrics(topology_name, metric_stats, interval=interval)
+
+                    # only report this once.
+                    if topology_status is None and _g(stats, 'unknown', None, 'status') is not None:
+                        topology_status = _g(stats, 'unknown', None, 'status').upper()
+                        if topology_status != 'ACTIVE':
+                            check_status = AgentCheck.CRITICAL
+                            self.service_check(
```

Comment:
I still think the 2 calls to `self.service_check` should be only one and the if/else should only set the value of `check_status`.